brownie/project/main.py

```diff
@@ -223,7 +224,37 @@ def load(self) -> None:
         self._active = True
         _loaded_projects.append(self)
 
-    def _get_changed_contracts(self, compiled_hashes: Dict) -> Dict:
+    def _flatten_contract(self, build_project: Build) -> str:
+        flatterner = "flatten_contracts.sol"
+        flatten_path = self._path.joinpath(flatterner)
+
+        base_contract_name = ""
+        has_dependencies = 0
+        for contract_name, _ in build_project.items():
+            contract_dependencies_length = len(build_project.get_dependents(contract_name))
+            if contract_dependencies_length == has_dependencies:
+                base_contract_name = contract_name
+        base_contract = build_project._sources._contracts[base_contract_name]
```

Comment:
The `Sources` object has a `get` method to access to `_contracts` and `_source`, so you don't have to use the private members directly.

Reply:
Ohhhh, I see.